src/cmdstan/command_helper.hpp

```diff
@@ -363,74 +468,75 @@ void get_theta_hat_csv(const std::string &fname,
       break;
     }
   }
-  for (size_t i = 0; i < param_names.size(); ++i) {
-    if (param_names[i].compare(names[i + col_offset]) != 0) {
+
+  // extract constrained parameter values
+  std::vector<double> cparams(cparam_names.size());
+  for (size_t i = 0; i < cparam_names.size(); ++i) {
+    if (cparam_names[i].compare(names[i + col_offset]) != 0) {
       msg << "Mismatch between model params and StanCSV file \"" << fname
-          << "\",  expecting param \"" << param_names[i] << "\", found \""
+          << "\",  expecting param \"" << cparam_names[i] << "\", found \""
           << names[i + col_offset] << "\"" << std::endl;
       throw std::invalid_argument(msg.str());
     }
     try {
-      theta_hat[i] = std::stod(values[i + col_offset]);
+      cparams[i] = std::stod(values[i + col_offset]);
     } catch (const std::exception &e) {
-      msg << "Error parsing CSV file, bad value for " << param_names[i]
+      msg << "Error parsing CSV file, bad value for " << cparam_names[i]
           << std::endl;
       throw std::invalid_argument(msg.str());
     }
   }
+  std::vector<double> uparams = unconstrain_params(model, cparams);
+  return Eigen::Map<Eigen::VectorXd>(&uparams[0], uparams.size());
```

Comment:
I'm not a C++ expert, is this implicitly returning a reference to a temporary value? It seems like something like this should lead to an invalid pointer later on, but maybe some memory magic is happening I don't understand.

Reply:
My first google hit suggests this could indeed be a problem: https://stackoverflow.com/questions/38786856/returning-a-eigenmap-from-a-lambda - Eigen maps do not own their data